dspace-api/src/main/java/org/dspace/authenticate/PasswordAuthentication.java

```diff
@@ -136,7 +137,9 @@ public boolean isImplicit()
 		// ensures they are password users
 		try
 		{
-			if (EPersonServiceFactory.getInstance().getEPersonService().getPasswordHash(context.getCurrentUser()) != null && !EPersonServiceFactory.getInstance().getEPersonService().getPasswordHash(context.getCurrentUser()).toString().equals(""))
+            if (context.getCurrentUser() != null &&
+                    (EPersonServiceFactory.getInstance().getEPersonService().getPasswordHash(context.getCurrentUser()) != null &&
```

Comment:
StringUtils.isNotBlank should safely treat null as blank, so this test isn't needed, saving us from reading one looooong line.